Проектная работа 2

Bjjyx opened the pull request:
Поставленные задачи выполнены.

sbuvaka-fhl committed:
Add dev branch for Y.Practicum

Bjjyx committed:
Произведено обобщение кода для более удобной работы с ним. Добавлен интерфейс ListDiffInterface. Классы ContactUi, FilterContactTypeUi и SortTypeUI реализуют данный интерфейс. Добавлен общий класс BaseListDiffCallback.

The code has been generalized for more convenient work with it. Added the ListDiffInterface interface. The Contactus, FilterContactTypeUi, and SortTypeUI classes implement this interface. Added a common BaseListDiffCallback class.

kgornostaeva commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.hashCode() == newItem.hashCode();
```
⚠️ Здесь должна использоваться функция theSameAs из ListDiffInterface

kgornostaeva commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/ListDiffInterface.java:
```diff
@@ -0,0 +1,7 @@
+
+public interface ListDiffInterface<T> {
+
+    boolean theSameAs(T t);
```
🍏 Рекомендую давать аргументам/переменным/параметрам развернутые имена, чтобы потом легко было понять, что здесь ожидается.
Например, newItem, other

kgornostaeva replied:
⚠️ Метод не используется, а должен)

kgornostaeva commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java:
```diff
@@ -20,6 +22,11 @@ public boolean isSelected() {
 
+    @Override
+    public boolean theSameAs(FilterContactTypeUi newFilterContactTypeUi){
+        return this.hashCode() == newFilterContactTypeUi.hashCode();
```
⚠️ В удаленном коде была другая реализация
```
oldItem.getContactType() == newItem.getContactType();
```

kgornostaeva commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/sort/SortTypeUI.java:
```diff
@@ -22,6 +23,11 @@ public boolean isSelected() {
 
+    @Override
+    public boolean theSameAs(SortTypeUI newSortTypeUi){
+        return this.hashCode() == newSortTypeUi.hashCode();
```
⚠️ В удаленном коде была другая реализация
```
oldItem.getSortType() == newItem.getSortType();
```

Bjjyx committed:
Исправлены ошибки в коде:
В классе DaseListDiffCallback переделан метод areItemsTheSame(), теперь он вызывает метод theSameAs() реализуемый классами от интерфейса ListDiffInterface.
Исправлены ошибки в методе theSameAs() в классах FilterContactTypeUi и SortTypeUI

Fixed bugs in the code:
In the DaseListDiffCallback class, the areItemsTheSame() method has been redesigned, now it calls the theSameAs() method implemented by classes from the ListDiffInterface interface.
Fixed errors in the theSameAs() method in the FilterContactTypeUi and SortTypeUI classes
